### PR TITLE
Jc view reform

### PR DESCRIPTION
**Pull request by jasondcater**
This reforms most of the view code, manifests and plots.

### CODE REVIEW

**Comment by graft** [comment]
Issues I have found:

1. On the Project page for IPI, no custom view template is defined (or does not load).

2. Markdown rendering in the 'browse' view does not work for certain attributes (the FAQ, e.g.)

3. On the Manifest page, there appear to be some styling issues (tested in Firefox 57):
  - the title font is a faint gray, not very visible, as is the radio button
  - horizontal rules are abutting against the buttons
  - Clicking 'edit' does not consistently preserve box or font sizes, so elements jump around
  - There is no more 'test' button while editing a script. Saving the script does not bust the consignment cache, so you cannot see new results from editing without reloading the page.
  - Running the 'Fingerprint' manifest fails with something about 'list_items[idx]'

4. On the plot page, I simply cannot create a new plot. I select a manifest and the 'Plot select' manifest remains unselectable.

File: app/models/archimedes/function.rb [hunk 1]
```diff
@@ -15,20 +17,20 @@ def all_functions
         ObjectSpace.each_object(::Class).select {|k| k < self }
       end
 
-      def is_func? function_name
-        self.name.split(/::/).last.snake_case == function_name.to_s
+      def is_func?(function_name)
+        self.name.split(/::/).last.underscore == function_name.to_s
```
Comment (graft) [inline app/models/archimedes/function.rb]: This is actually from Rails' ActiveSupport gem, which is also widely used outside of rails (e.g. I think bundler depends on it). String#snake_case and String#camel_case actually come from the gem 'extlib', which I use a lot just for this purpose.

Comment (jasondcater) [reply app/models/archimedes/function.rb]: As long as it's not rails specific. I'll add it back in and update the gem files where 'extlib' was missing.

File: db/migrate/20171107000013_change_view_models.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
+class ChangeViewModels < ActiveRecord::Migration
+  def change
+
+    drop_table :view_attributes
```
Comment (graft) [inline db/migrate/20171107000013_change_view_models.rb]: Can you amend this migration so it does not drop tables? If you are having 'not null' issues, you can set a default and drop it later, e.g.:

alter_table :something do
  add_column :whatever, null: false, default: 'etc'
end
alter_table :something do
  set_column_default :whatever, nil
end

**Comment by graft** [comment]
Some issues:

- The checkmark on Timur's message interface to dismiss a message does not work.

**Comment by jasondcater** [comment]
During syntax cleanup of the file `magma.js` the function called `model_names()` was changed to `modelNames()`. The files `search.jsx` and `model_map.jsx` still call the old name. We need to update the references in those two files.

**Comment by jasondcater** [comment]
Need to add in `PlotMatrixResult` from the `da-plot-heatmap` branch.

**Pull request merged**